### DISCUSSION TITLE
About `crate-type = ["staticlib"]`

### DISCUSSION THREAD

**Discussion by Jhonfunk** [fzyzcjy/flutter_rust_bridge#1774] [Q&A]
It would be nice if `flutter_rust_bridge` added the option to have rust **only generate static libraries** (using the `ffi.DynamicLibrary.executable` load symbol within flutter).

**Comment by fzyzcjy** [comment]
Hi, firstly, may I know why do you need that?

In addition, flutter_rust_bridge does not restrict any ways of building/linking. Thus, it is possible to tweak the underlying cargokit for your need.

**Reply by Jhonfunk** [reply]
On desktop OS, it is sometimes necessary to protect cryptographic algorithms using VMProtect. In case of using dynamic link libraries, symbols are always exposed. Using VMProtect after linking a static library to the main program is a bit safer（Even if the parameters of the symbols are figured out, you can't simply take the dynamic library and use it directly）.

**Reply by fzyzcjy** [reply]
I see. Feel free to ask questions if the proposed methods above does not work!
